timelock-impl/src/test/java/com/palantir/atlasdb/timelock/paxos/AcceptorCacheImplTests.java

```diff
@@ -0,0 +1,164 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.paxos;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+public class AcceptorCacheImplTests {
+
+    @Test
+    public void unseenCacheKeyThrows() {
+        AcceptorCache cache = cache(ImmutableMap.of());
+        AcceptorCacheKey cacheKey = AcceptorCacheKey.newCacheKey();
+
+        assertThatExceptionOfType(InvalidAcceptorCacheKeyException.class)
+                .as("we've not seen this cache key before, so we throw")
+                .isThrownBy(() -> cache.updatesSinceCacheKey(cacheKey));
+    }
+
+    @Test
+    public void cacheKeyWithNoUpdatesReturnsEmpty() throws InvalidAcceptorCacheKeyException {
+        Map<Client, Long> before = ImmutableMap.<Client, Long>builder()
+                .put(Client.of("client1"), 5L)
+                .put(Client.of("client2"), 6L)
+                .put(Client.of("client3"), 7L)
+                .build();
+        AcceptorCache cache = cache(before);
+
+        AcceptorCacheDigest digest = cache.getAllUpdates();
+
+        assertThat(digest)
+                .extracting(AcceptorCacheDigest::updates)
+                .as("get all the updates")
+                .isEqualTo(before);
+
+        AcceptorCacheKey newCacheKey = digest.newCacheKey();
+        assertThat(cache.updatesSinceCacheKey(newCacheKey))
+                .as("we have not added anything after we asked for updates initially")
+                .isEmpty();
+    }
+
+    @Test
+    public void updatingWithNonRecentValueDoesNotAffectCache() throws InvalidAcceptorCacheKeyException {
+        Map<Client, Long> before = ImmutableMap.<Client, Long>builder()
+                .put(Client.of("client1"), 5L)
+                .put(Client.of("client2"), 6L)
+                .put(Client.of("client3"), 7L)
+                .build();
+        AcceptorCache cache = cache(before);
+        AcceptorCacheDigest digest = cache.getAllUpdates();
+
+        cache.updateSequenceNumbers(ImmutableSet.of(
+                WithSeq.of(10L, Client.of("client1")),
+                WithSeq.of(3L, Client.of("client2"))));
+
+        AcceptorCacheDigest secondDigest = cache.updatesSinceCacheKey(digest.newCacheKey()).get();
+        assertThat(secondDigest.updates())
+                .containsEntry(Client.of("client1"), 10L);
+
```

Comment:
`assertThat(secondDigest.updates().size()).isEqualTo(1);`

Reply:
changed to `containsOnly`